CONTRIBUTING.md

```diff
@@ -0,0 +1,19 @@
+# Contributing
```

Comment:
All other meson-python documentation is in reST format. This should be too.

Reply:
The only thing that the GitHub native support for `CONTRIBUTING.md` does is to show the content when a new pull request is opened. I argue that the proposed content is only relevant to help contributors in the stage of development that come before the point of preparing a PR, thus I see the GitHub native support for `CONTRIBUTING.md` only showing the added documentation in the wrong context. I much prefer having all the documentation in a uniform format and, eventually, being able to link it to the rendered HTML documentation.